mlflow/store/artifact/azure_blob_artifact_repo.py

```diff
@@ -109,6 +109,9 @@ def list_artifacts(self, path=None):
             else:  # Just a plain old blob
                 file_name = posixpath.relpath(path=r.name, start=artifact_path)
                 infos.append(FileInfo(file_name, False, r.size))
+        # If the path is directed at a single file, return an empty list.
+        if (len(infos) == 1) and (infos[0].is_dir is False):
```

Comment:
Sorry for missing this earlier
```suggestion
        if (len(infos) == 1) and (infos[0].is_dir is False) and (infos[0].path == path):
```
This is because if a directory contains a single file and list_artifacts is called with the `path=directory` , the function should return:
[<FileInfo: file_size=<some-size>, is_dir=False, path='<directory>/<file>'>].
However, without the path comparison, the function would return an empty list, which would be incorrect.
(Thanks @dbczumar for pointing this out)

Reply:
@arjundc-db Are we sure that `path` is the correct right-hand-side for this comparison? Are we sure it isn't `dest_path`?

Reply:
This doesn't work as the `FileInfo.path` has been stripped of its root path, which needs to be compared to `dest_path`.
The logic can be done before creation of the `FileInfo` object, as shown in https://github.com/mlflow/mlflow/pull/3014/commits/b628d2db4d76823ed3b50ff90aef3ef4d56e17ec
(At this point in the code we know it's not a directory, and if the path has an exact match there can't be multiple files)

Reply:
Sorry, I was overthinking this. Reverted back to @dbczumar 's suggestion and added the relative path:
https://github.com/mlflow/mlflow/pull/3014/commits/28be18c5c903250a186919bfb43c33ca27722071